ldn-access-request.md

```diff
@@ -0,0 +1,103 @@
+# Requesting Access with Linked Data Notifications
+
+Status: draft
+
+
+## Abstract
+This proposal describes an exchange of Linked Data Notifications with a specific shape
+for the purpose of requesting access to a resource.
+
+
+## Actors
+- **Resource server:**
+  server on which the resource is located
+- **Requesting agent:**
+  agent requesting access to the resource
+- **Processing agent:**
+  agent accepting or rejecting the access request
+
+
+## Request flow
+
+1. The requesting agent assumes it does not have access to a given resource `R`.
+   For instance, it might have received a `403` in response to `GET` request,
+   or a `404` for a resource it knows exists.
+
+2. The requesting agent discovers an _access request inbox_ of `R`,
+   of from an agent that has access to `R`.
+   This might have been communicated out-of-band,
+   or can be linked with a `tdb:accessRequestInbox` relation,
+   or with a generic `ldp:inbox` relation.
+
+3. The requesting agent sends an _access request_
+   conforming to the shape described below
+   using the [Linked Data Notifications](https://www.w3.org/TR/ldn/) Protocol.
+
+4. The processing agent _optionally_ (depending on the request)
+   sends an _accept_ or _reject_ notification to an inbox of the requesting agent.
+
+5. The requesting agent retries accessing the resource.
+
+
+## Request access shape
+
+_This example needs to be formalized into a SHACL and/or ShEx shape._
```

Comment:
Quick guess:
```turtle
PREFIX acl: <http://www.w3.org/ns/auth/acl#>
PREFIX as: <https://www.w3.org/ns/activitystreams#>
PREFIX tbd: <http://example.org/to-be-determined#>


tbd:ReqShape {
  a [tbd:AppendRequest]? ;
  as:actor IRI ;
  as:target IRI ;
  as:object {
    a [acl:Authorization] ;
    acl:agent IRI ;
    acl:accessTo IRI ;
    acl:mode [acl:Read acl:Write acl:Whatever]+
  }
}
```

\[[try it](http://shex.io/webapps/shex.js/doc/shex-simple?schema=PREFIX%20acl%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fauth%2Facl%23%3E%0APREFIX%20as%3A%20%3Chttps%3A%2F%2Fwww.w3.org%2Fns%2Factivitystreams%23%3E%0APREFIX%20tbd%3A%20%3Chttp%3A%2F%2Fexample.org%2Fto-be-determined%23%3E%0A%0A%0Atbd%3AReqShape%20%7B%0A%20%20a%20%5Btbd%3AAppendRequest%5D%3F%20%3B%0A%20%20as%3Aactor%20IRI%20%3B%0A%20%20as%3Atarget%20IRI%20%3B%0A%20%20as%3Aobject%20%7B%0A%20%20%20%20a%20%5Bacl%3AAuthorization%5D%20%3B%0A%20%20%20%20acl%3Aagent%20IRI%20%3B%0A%20%20%20%20acl%3AaccessTo%20IRI%20%3B%0A%20%20%20%20acl%3Amode%20%5Bacl%3ARead%20acl%3AWrite%20acl%3AWhatever%5D%2B%0A%20%20%7D%0A%7D%0A&data=PREFIX%20acl%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fauth%2Facl%23%3E%0APREFIX%20as%3A%20%3Chttps%3A%2F%2Fwww.w3.org%2Fns%2Factivitystreams%23%3E%0APREFIX%20tbd%3A%20%3Chttp%3A%2F%2Fexample.org%2Fto-be-determined%23%3E%0A%5B%0A%20%20a%20tbd%3AAppendRequest%3B%0A%20%20%20%20as%3Aactor%20%3Chttps%3A%2F%2Fuser%2F%23me%3E%3B%0A%20%20%20%20as%3Atarget%20%3Chttps%3A%2F%2Fresources%2Fdocuments%2Fx.acl%3E%3B%0A%20%20%20%20as%3Aobject%20%5B%0A%20%20%20%20%20%20a%20acl%3AAuthorization%3B%0A%20%20%20%20%20%20acl%3Aagent%20%3Chttps%3A%2F%2Fuser%2F%23me%3E%3B%0A%20%20%20%20%20%20acl%3AaccessTo%20%3Chttps%3A%2F%2Fresources%2Fdocuments%2Fx%3E%3B%0A%20%20%20%20%20%20acl%3Amode%20acl%3ARead%2C%20acl%3AWrite%3B%0A%20%20%20%20%5D%3B%0A%5D.%0A&manifestURL=http%3A%2F%2Fshex.io%2Fwebapps%2Fshex.js%2Fexamples%2Fmanifest.json&shape-map=%7BFOCUS%20a%20tbd%3AAppendRequest%7D%40tbd%3AReqShape&interface=minimal&success=proof&regexpEngine=eval-threaded-nerr)\]